src/main/java/nl/tudelft/ti2206/group9/level/CollisionHandler.java

```diff
@@ -38,28 +43,40 @@ public void collide(final AbstractEntity collider,
      */
     public CollisionMap defaultCollisions() {
         final CollisionMap collisionMap = new CollisionMap();
-
         collisionMap.onCollision(Player.class, AbstractObstacle.class,
                 (collider, collidee) -> {
-                    if (!PowerupInvulnerable.isActive()) {
-                        OBSERVABLE.notify(
-                                GameObserver.Category.PLAYER,
-                                GameObserver.Player.COLLISION,
-                                AbstractObstacle.class.getSimpleName());
+                    if (!AbstractPowerup.isActive(PowerupInvulnerable.class)) {
                         collider.die();
+                        notifyCollide(AbstractObstacle.class);
+                    }
+                });
+        collisionMap.onCollision(Player.class, Log.class,
+                (collider, collidee) -> {
+                    if (AbstractPowerup.isActive(PowerupDestroy.class)) {
+                        collidee.selfDestruct();
+                        notifyCollide(Log.class);
+                        State.addScore(Log.VALUE);
+                        return;
```

Comment:
Oh wait i see. Can't you just replace the if statement following this method with an else if statement?